app/services/DataService.scala

```diff
@@ -13,7 +13,20 @@ class DataService @Inject() () {
     database += record
   }
 
-  def search(maybeName: Option[String] = None): ListBuffer[DataModel] = {
-    maybeName.fold(database)(name => database.filter(_.name.equalsIgnoreCase(name)))
+  def search(maybeName: Option[String] = None, maybeAge: Option[String], maybeMessage: Option[String]): ListBuffer[DataModel] = {
+    val cleansedMaybeName = convertEmptyInputToNone(maybeName)
+    val cleansedMaybeAge = convertEmptyInputToNone(maybeAge)
+    val cleansedMaybeMessage = convertEmptyInputToNone(maybeMessage)
+
+    val predicates: List[DataModel => Boolean] = List(
+      cleansedMaybeName.map(name => (record: DataModel) => record.name.equalsIgnoreCase(name)),
```

Comment:
if you wanted you could in-line the filter(_.nonEmpty), i.e.
`maybeName.filter(_.nonEmpty).map( name => (record: DataModel) => record.name.equalsIgnoreCase(name))` etc